src/components/StopWatch.jsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useRef, useState } from "react";
+import Pause from "../assets/pause.png";
+import Play from "../assets/play.png";
+import Restart from "../assets/restart.png";
+import Stop from "../assets/stop.png";
+import PauseDark from "../assets/pauseDark.png";
+import PlayDark from "../assets/playDark.png";
+import RestartDark from "../assets/restartDark.png";
+import StopDark from "../assets/stopDark.png";
+import { useTasks } from "../context/task-context";
+import { useTheme } from "../context/theme-context";
+
+export const StopWatch = ({ duration, taskId, disabled }) => {
+  const { completeTask } = useTasks();
+  const { theme } = useTheme();
+  const {dark} = theme;
+  const timerRef = useRef({ timerId: null, currentDuration: null });
+  const totalSeconds = timerRef.currentDuration
+    ? timerRef.currentDuration
+    : duration.hours * 60 * 60 + duration.minutes * 60 + duration.seconds;
+  const [progress, setProgress] = useState(totalSeconds);
+  const [isPaused, setIsPaused] = useState(true);
+
+  const start = () => {
+    timerRef.current.timerId = setInterval(() => {
+      setIsPaused(false);
```

Comment:
does setIsPaused need to be updated every second ? or only once. if once, it should be outside interval, even though it doesn't trigger re render, it is not good coding practice.